plugins/CuraEngineBackend/CuraEngineBackend.py

```diff
@@ -1094,14 +1096,18 @@ def disableTimer(self) -> None:
             self._change_timer.timeout.disconnect(self.slice)
 
     def _onPreferencesChanged(self, preference: str) -> None:
-        if preference != "general/auto_slice" and preference != "info/send_engine_crash":
+        if preference != "general/auto_slice" and preference != "info/send_engine_crash" and preference != "info/anonymous_engine_crash_report":
             return
         if preference == "general/auto_slice":
             auto_slice = self.determineAutoSlicing()
             if auto_slice:
                 self._change_timer.start()
         elif preference == "info/send_engine_crash":
-            os.environ["use_sentry"] = "1" if CuraApplication.getInstance().getPreferences().getValue("info/send_engine_crash") else "0"
+            os.environ["USE_SENTRY"] = "1" if CuraApplication.getInstance().getPreferences().getValue("info/send_engine_crash") else "0"
+        elif preference == "info/anonymous_engine_crash_report":
+            account = CuraApplication.getInstance().getCuraAPI().account
+            if account and account.isLoggedIn and not CuraApplication.getInstance().getPreferences().getValue("info/anonymous_engine_crash_report"):
```

Comment:
You mean like this? that will add an extra indent also it will only check if the preference changes
```
        elif preference == "info/anonymous_engine_crash_report":
            account = CuraApplication.getInstance().getCuraAPI().account
            if not CuraApplication.getInstance().getPreferences().getValue("info/anonymous_engine_crash_report"):
                account = CuraApplication.getInstance().getCuraAPI().account
                if account and account.isLoggedIn:
```